pipeline/tasks/publish_tasks.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tasks for publishing artman output"""
+
+import subprocess
+from pipeline.tasks import task_base
+
+
+class DevpiUploadTask(task_base.TaskBase):
+    """Publishes to a devpi index"""
+
+    def execute(self, pypi_server_url, pypi_uname, pypi_pwd, publish_env,
+                final_repo_dir):
+        publish_url = pypi_server_url + pypi_uname + '/' + publish_env
+        subprocess.check_call(
+            ['devpi',
+             'login',
```

Comment:
This might be a limitation of devpi, which doesn't allow passing username/password (or auth token) during publishing. That makes the implementation looks like logging in every time the task gets running.

On the other hand, if we don't do the login in this task, it would mean that whoever runs the pipeline for pypi publishing need to do `devpi login` manually. I feel that is not ideal either.